### PR TITLE
test/misc/fix-e2e-handle-push-perm-request : Fix end-to-end to handle push notification permission request dialog

### LINKED ISSUE #235: Fix end-to-end to handle push notification permission request dialog
**Is your feature request related to a problem? Please describe.**
With PR #234, a new dialog pops-up when arriving on the Profile screen for the first time. This makes the end-to-end tests fail.

**Describe the solution you'd like**
We should add this at the top of the end-to-end test classes:
```kotlin
@get:Rule
  val permissionRule: GrantPermissionRule =
      GrantPermissionRule.grant(Manifest.permission.POST_NOTIFICATIONS)
```

**Additional context**
It makes it so that that permission to push notifications is granted, so the dialog is never shown.
I will create an issue to update them in their branch.

### PR DESCRIPTION
**Pull request by charliemangano**
# Fix end-to-end to handle push notification permission request dialog

**This PR, contrary to usual, is asking to merge into branch `test/end-to-end` and not `main`.**

## Description
This PR introduces introduces a fix to issue #235, where the dialog requesting push notification permissions was making the end-to-end tests fail. More details in the description of the issue.
**Note:** The only commit relevant for review is [commit 85c39ab](https://github.com/PeriodPals/periodpals/commit/85c39abde9cf5bd03f8d6227386cb6f07647608a). For some reason, every commit pushed to `main` since the last update of `test/end-to-end` was rebased onto this branch instead of merged, I promise I didn't make 154 commits modifying 55 files. I'll be more careful with my git manipulations next time.

## Changes
Added test `Rule` that grants the permission by default, so that the dialog is never shown.

#### Added
None
#### Modified
- `app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndM1.kt`
- `app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndM2.kt`

#### Removed
None

## Dependencies Added
None

## Testing
Tests now pass. I have not modified their behavior other than was said above.

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/model/alert/Alert.kt [hunk 1]
```diff
@@ -16,7 +18,7 @@ package com.android.periodpals.model.alert
  */
 data class Alert(
     val id: String?, // given when created in supabase
-    val uid: String?,
```
Comment (coaguila) [inline app/src/main/java/com/android/periodpals/model/alert/Alert.kt:19]: Why are we changing this value to a non-billable value?

File: app/src/main/java/com/android/periodpals/model/alert/Alert.kt [hunk 2]
```diff
@@ -29,7 +31,8 @@ data class Alert(
 /** Enum class representing the product requested with the alert. */
 enum class Product {
   TAMPON,
-  PAD
+  PAD,
+  NO_PREFERENCE,
```
Comment (coaguila) [inline app/src/main/java/com/android/periodpals/model/alert/Alert.kt:35]: Great catch

**Review by coaguila — APPROVED** [review]
Lgtm

**Review by taghizadlaura — APPROVED** [review]
## Summary
LGTM :
This PR ensures end-to-end tests no longer fail due to the push notification permission request dialog. The implementation adds a test `Rule` that grants permissions by default, bypassing the dialog. 

## Important

### Code Quality
LGTM : The use of a test `Rule` is a clean and effective solution.

### Functionality
LGTM :
Tests now pass without modification to their original behavior. 

### Performance
No permission issues.

### Security
No security concerns.

## Testing
LGTM :
The tests now pass as expected after applying the fix.

## Documentation
LGTM :
The PR description provides sufficient context about the issue, the implemented solution, and its impact.

## Suggestions
None.

## Steps Before PR Approval
LGTM

**Pull request merged**